web/packages/teleport/src/Discover/Server/EnrollEc2Instance/EnrollEc2Instance.tsx

```diff
@@ -141,9 +142,14 @@ export function EnrollEc2Instance() {
       );
 
       const ec2InstancesLookupByInstanceId: Record<string, Node> = {};
-      fetchedNodes.forEach(
-        d => (ec2InstancesLookupByInstanceId[d.awsMetadata.instanceId] = d)
-      );
+      fetchedNodes.forEach(d => {
+        // Extract the instanceId of the fetched node from its label.
+        const instanceId = d.labels.find(
+          label => label.name === 'teleport.dev/instance-id'
+        )?.value;
+
+        ec2InstancesLookupByInstanceId[instanceId] = d;
```

Comment:
are we expecting this to be ever undefined? if there is apossibility, we probably shouldn't add it as `undefined`

Reply:
No since these results are being returned by the search with a predicate query which specifically queries for this label